dataflow/src/main/java/org/checkerframework/dataflow/cfg/node/ObjectCreationNode.java

```diff
@@ -29,9 +30,11 @@ public class ObjectCreationNode extends Node {
     /** The enclosing expression of the object creation or null. */
     protected final @Nullable Node enclosingExpression;
 
-    // TODO: See issue 376
-    /** The constructor node of the object creation. */
-    protected final Node constructor;
+    /**
+     * The identifier node of the object creation. A generic constructor identifier can refer to the
```

Comment:
It's a bit odd that you're referring to "constructor" here, even though you just renamed the field from constructor to identifier.
Maybe identifier isn't the best name, either? It is better, because `NewClassTree` calls it identifier. But since Java 5 it also contains the type arguments, for which "identifier" is misleading. But "type" would probably also be confusing.

Reply:
What kind of `Node` is it for annotated types, e.g. in `new @Foo Object()` or `new @Bar ArrayList<String>()`?

Reply:
Same, new @Foo Object() refers to `ClassNameNode` and `new @Bar ArrayList<String>()` refers to `ParameterizedTypeNode`